### ISSUE TITLE
remote_graph.push() not reporting an error when a Neo4J constraint is involved?

### ISSUE THREAD

**Issue by tonyennis** [nigelsmall/py2neo#419]
In neo4J, I have this constraint:

CREATE CONSTRAINT ON (v:VERSION) ASSERT v.version IS UNIQUE

I run this:

```python
import time

from py2neo import Node
from app import remote_graph

def make():
    v = Node("VERSION")
    v.properties['version'] = time.time()
    remote_graph.create(v)
    remote_graph.push(v)
    return v

"""
First, make an empty Neo4J database and run this:

CREATE CONSTRAINT ON (v:VERSION) ASSERT v.version IS UNIQUE

"""


v1 = make()
v2 = make()

print v1
print v2

# This is an invalid situation
v2.properties['version'] = v1.properties['version']

# This should be rejected with a constraint failure
remote_graph.push(v2)

# But there is no evidence of rejection here
print v1
print v2

# If you check Neo4J, you see the version values are different.
```

When I set v2's version to the same value as v1, I expect the push(v2) to fail. It does fail, but nothing is reported.  Compare v1 and v2 at the end with the values in Neo4J.

I got this for the program:

    (n322232:VERSION {version:1436885714.82507})
    (n322233:VERSION {version:1436885714.844348})
    (n322232:VERSION {version:1436885714.82507})
    (n322233:VERSION {version:1436885714.82507})

and this when looking at Neo4J with their web tool (cleaned up, copy/paste went splat):

    $MATCH (n) RETURN id(n), n;
    id(n)                 version
    322232		1436885714.82507
    322233		1436885714.844348

**Comment by nigelsmall** [comment]
Can you clarify which version of Neo4j you're using?

**Comment by tonyennis** [comment]
2.2.0, according to the 'greeting' page in the Neo4J web tool.

**Comment by nigelsmall** [comment]
This should be fixed by 370dbf07f41e650ba8d638a4c7e338be83cf0577 and released in 2.0.8.

**Comment by tonyennis** [comment]
Rawk
